src/components/GithubLeaderboard.tsx

```diff
@@ -0,0 +1,120 @@
+import { Flex, Spinner, Text, Box, useToast } from '@chakra-ui/react';
+import React, { useEffect } from 'react';
+import { GitHubOrderBy, GitHubStats } from '../types';
+import Leaderboard from './Leaderboard';
+import { OrderBySelect } from './OrderBySelect';
+import { useState } from 'react';
+import { getGitHubLeaderboard } from '../services/leaderboard';
+import { lastUpdated } from '../utils';
+
+interface Props {}
+
+export const GithubLeaderboard: React.FC<Props> = () => {
+  const [githubOrder, setGithubOrder] = useState<GitHubOrderBy>(
+    GitHubOrderBy.Commits
+  );
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string>();
+  const [githubData, setGithubData] = useState<GitHubStats[]>([]);
+
+  const toast = useToast();
+
+  const fetchData = async () => {
```

Comment:
doesn't need to be in this refactor, but i wonder if we can move leaderboard fetching and state into a generic hook. then i'm sure even more simplification/factoring will follow

Reply:
Yeah we definitely could. There are some other opportunities for cleaning up code in the `Leaderboard` component, especially regarding how each cell in the leaderboard table is made.